### PR TITLE
test: add unit tests for sequence variability module

### LINKED ISSUE #35: Add unit tests for Sequence Variability Module
This issue is for adding unit tests to the `Sequence Variability` module.

### PR DESCRIPTION
**Pull request by aniketfadia96**
Hi, I have made the following changes in this PR -

- Add unit tests for `Sequence Variability Module`
- Add edge case for empty dataframe conditions

Closes #35

### CODE REVIEW

File: tests/genetic_testing/sequence_analysis/test_sequence_variability.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        MIN_COUNT: 2,
+        EXPECTED_RESULT: pd.DataFrame(
+            {"Sequence": ["ACGT", "TGCA"], "Count": [2, 3], "Group_Number": [1, 2]},
+        ),  # Test case for multiple sequences belonging to different groups
+    },
+}
+
+
+@pytest.mark.parametrize(
+    "sequences, base_sequence, min_count, expected_result",
```
Comment (lsetiawan) [inline tests/genetic_testing/sequence_analysis/test_sequence_variability.py:64]:
```suggestion
    [SEQUENCES, BASE_SEQUENCE, MIN_COUNT, EXPECTED_RESULT],
```

File: tests/genetic_testing/sequence_analysis/test_sequence_variability.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    [
+        (
+            data["sequences"],
+            data["base_sequence"],
+            data["min_count"],
+            data["expected_result"],
```
Comment (lsetiawan) [inline tests/genetic_testing/sequence_analysis/test_sequence_variability.py:70]:
```suggestion
            data[SEQUENCES],
            data[BASE_SEQUENCE],
            data[MIN_COUNT],
            data[EXPECTED_RESULT],
```

File: tests/genetic_testing/sequence_analysis/test_sequence_variability.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            "seq4": "TGCA",
+            "seq5": "TGC-",
+        },
+        BASE_SEQUENCE: "ACGT",
+        MIN_COUNT: 2,
+        EXPECTED_RESULT: pd.DataFrame(
+            {"Sequence": ["ACGT", "TGCA"], "Count": [2, 3], "Group_Number": [1, 2]},
+        ),  # Test case for multiple sequences belonging to different groups
+    },
+}
```
Comment (lsetiawan) [inline tests/genetic_testing/sequence_analysis/test_sequence_variability.py:60]: I have a question about this. Why is this a `Dict` of `Dict` instead of maybe a `Dict` of `tuple`? 

```python
EXPECTED_RESULTS = {
        "empty_sequences": ({}, "ACGT", 2, None),
}

# Then you can easily pass in the ids and such
@pytest.mark.parametrize(
    [SEQUENCES, BASE_SEQUENCE, MIN_COUNT, EXPECTED_RESULT],
    list(EXPECTED_RESULTS.values()),
    ids=list(EXPECTED_RESULTS.keys())
)
```

Dict nowadays keeps its order so it should be consistent when you call `.keys` and `.values`

Comment (aniketfadia96) [reply tests/genetic_testing/sequence_analysis/test_sequence_variability.py:60]: Okay, I like your approach. I thought that it will be more readable if I use a `Dict` of `Dict` so that each input parameter has a key associated with it. But in this case, since we have only 4 parameter, we can use `Dict` of `Tuple`.

**Review by lsetiawan — APPROVED** [review]
LGTM
